api/src/OpenTelemetry/Internal/Logging/Types.hs

```diff
@@ -103,44 +196,37 @@ data LogRecord body = LogRecord
   --    - A byte array,
   --    - An array (a list) of any values,
   --    - A map<string, any>.
-  , logRecordResource :: MaterializedResources
-  -- ^ Describes the source of the log, aka resource. Multiple occurrences of events coming from the same event source can happen across time and they all have the same value of Resource.
-  -- Can contain for example information about the application that emits the record or about the infrastructure where the application runs. Data formats that represent this data model
-  -- may be designed in a manner that allows the Resource field to be recorded only once per batch of log records that come from the same source. SHOULD follow OpenTelemetry semantic conventions for Resources.
-  -- This field is optional.
-  , logRecordInstrumentationScope :: InstrumentationLibrary
   , logRecordAttributes :: LogAttributes
   -- ^ Additional information about the specific event occurrence. Unlike the Resource field, which is fixed for a particular source, Attributes can vary for each occurrence of the event coming from the same source.
   -- Can contain information about the request context (other than Trace Context Fields). The log attribute model MUST support any type, a superset of standard Attribute, to preserve the semantics of structured attributes
   -- emitted by the applications. This field is optional.
   }
-  deriving (Functor)
 
 
 {- | Arguments that may be set on LogRecord creation. If observedTimestamp is not set, it will default to the current timestamp.
 If context is not specified it will default to the current context. Refer to the documentation of @LogRecord@ for descriptions
 of the fields.
 -}
-data LogRecordArguments body = LogRecordArguments
+data LogRecordArguments = LogRecordArguments
   { timestamp :: Maybe Timestamp
   , observedTimestamp :: Maybe Timestamp
   , context :: Maybe Context
   , severityText :: Maybe Text
   , severityNumber :: Maybe SeverityNumber
-  , body :: body
+  , body :: AnyValue
   , attributes :: H.HashMap Text AnyValue
   }
 
 
-emptyLogRecordArguments :: body -> LogRecordArguments body
-emptyLogRecordArguments body =
+emptyLogRecordArguments :: LogRecordArguments
+emptyLogRecordArguments =
```

Comment:
Removed body because of the addition of `NullValue` and the removal of the type parameter to `LogRecordArguments`